### DISCUSSION TITLE
single panel 64x64

### DISCUSSION THREAD

**Discussion by JeremC82** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#541] [Q&A] [answered]
Hello, I am facing a problem.
I wrote a code which depending on the activation of a strap configures the mcu for two 64x32 panels or a single 64x64 panel. It works perfectly on the two 64x32 panels. On the other hand, on the 64X64 model I have a big offset. However, I declared and connected my io 32 pin

`
*/
/*--------------------- MATRIX GPIO CONFIG  -------------------------*/
#define R1_PIN 25
#define G1_PIN 26
#define B1_PIN 27
#define R2_PIN 14
#define G2_PIN 12
#define B2_PIN 13
#define A_PIN 23
#define B_PIN 19
#define C_PIN 5
#define D_PIN 17
#define E_PIN 32 // required for 1/32 scan panels, like 64x64. Any available pin would do, i.e. IO32
#define LAT_PIN 4
#define OE_PIN 15
#define CLK_PIN 16

int PanelPin = 33;    // Strap panel pin mode 

void setup() {

  //clear NEOPIXEL
  pixels.clear();
  pixels.show();
  

  //init input button
  pinMode(TRIGGER_PIN, INPUT_PULLDOWN);
  pinMode(PanelPin, INPUT_PULLDOWN);    
  time_t t = 0;

  Serial.begin(115200);
  SPIFFS.begin(true);
  Serial.println(F("*****************************************************"));
  Serial.println(F("*        Detection du panneau                       *"));
  Serial.println(F("*****************************************************"));
  bool checkPanelPin;
  checkPanelPin = digitalRead(PanelPin);
  Serial.print("voici la configuration panel detecté =");
  if (checkPanelPin)
  {
    Serial.println("strap connecté");
    Serial.println("Configuration 1 Panel  64x64 ");
    // Configure for your panel(s) as appropriate!
    #define PANEL_RES_X 64  // Number of pixels wide of each INDIVIDUAL panel module. if dual 64 
    #define PANEL_RES_Y 64  // Number of pixels tall of each INDIVIDUAL panel module. if dual 64 
    #define PANEL_CHAIN 1  // si 1 panneau 

    #define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_BOTTOM_LEFT_UP
    //#define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_TOP_LEFT_DOWN
    //#define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_TOP_RIGHT_DOWN
    //#define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_BOTTOM_RIGHT_UP
  }
  else
  {
    Serial.println("strap non connecté ");
    Serial.println("Configuration 2 Panel  64x32 ");
    // Configure for your panel(s) as appropriate!
    #define PANEL_RES_X 64  // Number of pixels wide of each INDIVIDUAL panel module. if dual 64 
    #define PANEL_RES_Y 32  // Number of pixels tall of each INDIVIDUAL panel module. if dual 32 

    #define NUM_ROWS 2                      // Number of rows of chained INDIVIDUAL PANELS si deux panneaux = 2
    #define NUM_COLS 1                      // Number of INDIVIDUAL PANELS per ROW si deux panneaux = 1
    #define PANEL_CHAIN NUM_ROWS *NUM_COLS  // si plusieurs panneau

    //#define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_BOTTOM_LEFT_UP
    //#define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_TOP_LEFT_DOWN
    #define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_TOP_RIGHT_DOWN
    //#define VIRTUAL_MATRIX_CHAIN_TYPE CHAIN_BOTTOM_RIGHT_UP
  }

`

**Comment by JeremC82** [comment, marked as answer]
thanks for your help

**Reply by mrcodetastic** [reply]
Can you post a video of the issue?

**Reply by JeremC82** [reply]
yes , see attached https://youtube.com/shorts/hP3QO7u_qF0

**Reply by mrcodetastic** [reply]
Looks like some interference to me. Make you are not using GPIOs that are used for serial communications, also, make sure the wire running from the esp32 to panel is not longer than 10cm
